### ISSUE TITLE
PostgresUtils.toPGArray  does not handle escaped quotes

### ISSUE THREAD

**Issue by jandryse** [jOOQ/jOOQ#10467]
### Expected behavior and actual behavior:
```
PostgresUtils.toPGArray("{\"A\",\"B\\\"C\"}");
```
expected:
```
ImmutableList.of("A", "B\"C")
```
actual:
```
ImmutableList.of("A", "B\\")
```

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):
```
    @Test
    void canParseEscapedQuotes() {
        List<String> actual = PostgresUtils.toPGArray("{\"A\",\"B\\\"C\"}");
        assertThat(actual, is(ImmutableList.of("A", "B\"C")) );
    }
```

Actual code expects that quotes are escaped by another quotes.  Bu it is not the only possibility and in fact Postgresql usually uses the other possibility

```
select array['A','B"C']::TEXT;
   array
------------
 {A,"B\"C"}
(1 row)
```

### Versions:

- jOOQ:3.13.4
- Java: irrelevant
- Database (include vendor): irrelevant
- OS: irrelevant
- JDBC Driver (include name if inofficial driver): irrelevant

**Comment by lukaseder** [comment]
Thanks a lot for your report. I'll look into this soon

**Comment by lukaseder** [comment]
This seems to happen only when an array is embedded in a type. A plain array doesn't reproduce this:

```java
@Test
public void testPostgresArrayEscaping() throws Exception {
    bindAndInline(ctx -> {
        String[] expected = new String[] { "A", "B\"C" };
        String[] actual = ctx.fetchValue(val(expected));

        assertEquals(asList(expected), asList(actual));
    });
}
```

The call hierarchy shows this, too:

![image](https://user-images.githubusercontent.com/734593/89399920-707fde00-d713-11ea-8a4c-6f045d834870.png)

**Comment by lukaseder** [comment]
Yet, `DSL::rowField` also doesn't reproduce this case:

```java
Record1<String[]> r1 = ctx.fetchValue(rowField(row(val(expected))));
assertEquals(1, r1.size());
assertEquals(asList(expected), asList(r1.value1()));

Record2<String[], String[]> r2 = ctx.fetchValue(rowField(row(val(expected), val(expected))));
assertEquals(2, r2.size());
assertEquals(asList(expected), asList(r2.value1()));
assertEquals(asList(expected), asList(r2.value2()));
```

**Comment by lukaseder** [comment]
This helps reproduce it:

```sql
CREATE TYPE u_10467 AS (a text[])
```

And then:

```java
U_10467Record u = ctx.fetchValue(val(new U_10467Record(expected)));
assertEquals(asList(expected), asList(u.getA()));
```

**Comment by lukaseder** [comment]
This seems to also fix https://github.com/jOOQ/jOOQ/issues/6819, which is caused by the same issue.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.5 (#10471)
